web/tls_config.go

```diff
@@ -163,6 +234,11 @@ func ConfigToTLSConfig(c *TLSConfig) (*tls.Config, error) {
 		cfg.ClientCAs = clientCAPool
 	}
 
+	if c.ClientAllowedSanRegex.Regexp != nil {
+		// verify that the client cert contains the allowed domain name
+		cfg.VerifyPeerCertificate = c.VerifyPeerCertificate
```

Comment:
Why do we need that if?

Reply:
I guess I'm not fully tracking the question, without the `if` the behavior would be to always require the client to verify SAN; unless you are thinking we could default the regex to be fully permissive with something like `.*`?